pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTResource.java

```diff
@@ -48,24 +49,15 @@ public boolean isConciseResource() {
      * then returns the sequence of names that identifies the expression.
      * If this has a local variable declaration, then returns the name
      * of the variable.
+     *
+     * @deprecated Since 7.1.0. This method is not very useful because the expression
+     *          might be complex and not have a real "name". In this case we return the
+     *          expression pretty printed.
      */
+    @Deprecated
```

Comment:
`JavaAttributePrinter` will not print any deprecated attributes, so all tree dump tests using resources will fail with this change.

Reply:
Getters that return a node are not in the tree dump tests though iirc

Reply:
But this is not returning a node. It's a string. That's why the build is failing on this pr.

Reply:
The method `getStableName()` is also used in EmptyControlStatementRule (to exclude `unused` vars) - so, not sure whether we can simply deprecate this...

The javadoc is definitively useful to describe when to expect a "useful" value of stable name and when not...

Reply:
True, although in https://github.com/pmd/pmd/issues/4930 I argue we should stop caring about the name of the variable when it is a concise resource. So we don't need a "stable name" anymore, we just need to get the name of the VariableId when the resource is like a LocalVariableDeclaration